### ISSUE TITLE
Jstree 4

### ISSUE THREAD

**Issue by FredGr** [vakata/jstree#2286]
Hello 
Any update on Jstree 4 ?

Frederic.

**Comment by vakata** [comment]
Hi @FredGr 

I just published some dev code to:
https://github.com/vakata/jstree/tree/v.4.0

Some basic functionality is still missing, and the code works only in browsers that support generators, but that will change over the next few weeks.

**Comment by gpetrov** [comment]
v4 is looking great @vakata! It was about time :)
Hope to see the full functioning version soon - so we can put it in production!

**Comment by vakata** [comment]
@gpetrov glad to hear from you - flattered that you still follow the project! I will try to find some time for new v.4 features soon.

**Comment by ahmadalfy** [comment]
@vakata I love this project and I was asking if there's room to contribute. Where do you suggest I can start? Features, tests, docs or what?

**Comment by dixonnixon** [comment]
Hi. I need Help ;)
When Ive enabled both contextmenu and types plugins
with code
Hi. I need Help ;)
When Ive enabled both contextmenu and types plugins
with code:

`'contextmenu': { 
    items: function($node) { 
        var tree = $("#tree").jstree(true); 
        return { 
                "Create": { 
                        "separator_before": false, 
                        "separator_after": false, 
                        "label": "Cоздать", 
                        "action": function (obj) { 
                                $node = tree.create_node($node); tree.edit($node); 
                        } 
                }
        }` ...

tree.create_node inside action property didnt fired! Please Help Me.
When types plugin disabled its Ok

**Comment by yusufozturk** [comment]
Hi @vakata 

We are also very interested with new version. Currently there is no good tree library for react and it's good to know that you are dropping jquery dependency.

Do you might have any ETA?

Thanks.

**Comment by danielczestki** [comment]
Is it abandoned ?

**Comment by vakata** [comment]
No, jstree is not abandoned - I struggle to find the time to finish v.4. This is why I enabled my sponsors page - hoping some income from the project could help me buy more time. Anyway - it is coming, but I have no ETA.

**Comment by tacman** [comment]
I'm trying to integrate jstree with a stimulus controller (using webpack), and spending most of my time fighting with getting jQuery injected.

Of course, if v4 were out, I'd skip the jQuery part altogether.

Thanks for your work on this terrible library.

**Comment by vakata** [comment]
@lofcz thank you very much - I really appreciate this.

@tacman I am not forcing anyone to use jstree - if you do not like it - stop using it - there are plenty of alternatives. 
jsTree is not my full time job, in fact - the "profits" I make from jstree barely cover the domain name. Any time I spend on jstree comes out of my family time, or my work time - same as any small open source project. On one hand - I get your frustration and I feel for you (I wish I could do more, as I love this project), on the other hand - you are not talking to a team, or a corporation, or to a dev whose project pays the bills, nor is it something you pay for. jstree is provided to you as it is, completely free - it is your choice whether to use it or not. If you need assistance - be nice about it.

**Comment by tacman** [comment]
Oh, then that came off completely wrong!  There was zero intention to show
anything be appreciation for what you've done and enthusiasm for moving
away from jQuery.  If I communicated anything but that, please accept my
apologies.

I LOVE jstree.  I'm working on a article for Medium describing the best
practices for working with hierarchical data in Symfony, and jstree is
front and center.  A previous demo is at
https://jstree-symfony-demo.herokuapp.com/, the source is at
https://github.com/survos/jstree-demo.  In the process of updating it to
the latests components and best practices, I wanted to switch to stimulus,
and spent an hour yesterday fighting with the "jstree is not a function"
error. Obviously, this has nothing to do with jstree, I'm not configuring
jQuery correctly, thus the ping here to see the status of the
pure-javascript version.  I'm sure I'll eventually figure out the jQuery
issue.

Again, sorry if my comments came across wrong.

Tac

**Comment by vakata** [comment]
@tacman sorry for the rant, it seemed like a sarcastic "thanks" (because of the use of "terrible", which as it turns out was a typo). Thank you for clarifying :) As for v4 - I really am sorry for not being able to work on it as much as I want to. I am glad you like jstree, I hope v4 will be out soon!

**Comment by tacman** [comment]
I figured out my jQuery problem, I was confusing $.jstree(element) with
$(element).jstree().  DOH.

So I have it working locally now, I'll publish a demo using stimulus soon.
